CCompiler/Lexer.cs

```diff
@@ -0,0 +1,164 @@
+namespace CCompiler
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Text;
+  using System.Text.RegularExpressions;
+
+  public class Lexer
+  {
+    [Flags]
+    private enum LexerMode
+    {
+      Normal = 1 << 0,
+
+      Stirng = 1 << 1,
+      StirngEscape = 1 << 2,
+
+      CommentStart = 1 << 3,
+      Comment = 1 << 4,
+      CommentEed = 1 << 5,
+    }
+
+    private static readonly HashSet<char> endWordBreakers = new HashSet<char> { ' ', '\n', '\r' };
```

Comment:
It is not immedietly clear from the names what the difference between this set and the one below is. Perhaps this should be part of the wordBreakers set instead, and then have the subset thats in wordBreakers called charTokens or something similar

Reply:
just "wordBreakers" left now